demos/common/network_manager/aws_iot_network_manager.c

```diff
@@ -106,8 +110,8 @@ typedef struct NMNetworkInfo
 typedef struct NetworkManagerInfo
 {
     SemaphoreHandle_t xSubscriptionLock;
-    Link_t xSubscriptionListHead;
-    Link_t xNetworkListHead;
+    IotListDouble_t xSubscriptionListHead;
+    IotListDouble_t xNetworkListHead;
     IotTaskPool_t* pxTaskPool;
```

Comment:
I guess you are using the task pool pointer to initialize network maanger to use one specific task pool, correct?  if so, that is OK and you should use the system task pool via the handle #define'd in the task pool types header...
the max concurrent jobs is useless IMO, we should remove it. Just create dynamic tasks and recycle them in the callback.

Reply:
Updated in PR https://github.com/aws/amazon-freertos/pull/462